w07h02/test/pgdp/LightPanelTest.java

```diff
@@ -0,0 +1,503 @@
+
+        testclass.up("[SC]");
+        Assertions.assertFalse(testclass.down(arg));
+        assertEqualColorAndLevel("[SC]", 3, testclass);
+
+        testclass.up("red");
+        Assertions.assertFalse(testclass.down(arg));
+        assertEqualColorAndLevel("red", 4, testclass);
+
+        testclass.up("end");
+        Assertions.assertFalse(testclass.down(arg));
+        assertEqualColorAndLevel("yellow", 5, testclass);
+    }
+
+    @Test
+    @DisplayName("Blue")
+    void downBlue() {
+        String arg = "blue";
+
+        Assertions.assertFalse(testclass.down(arg));
+        assertEqualColorAndLevel("", 0, testclass);
+
+        testclass.up("green");
+        Assertions.assertFalse(testclass.down(arg));
+        assertEqualColorAndLevel("green", 1, testclass);
+
+        testclass.up("blue");
+        Assertions.assertTrue(testclass.down(arg));
+        assertEqualColorAndLevel("", 0, testclass);
+
+        testclass.up("yellow");
+        Assertions.assertFalse(testclass.down(arg));
+        assertEqualColorAndLevel("yellow", 2, testclass);
+
+        testclass.up("doubleYellow");
+        Assertions.assertFalse(testclass.down(arg));
+        assertEqualColorAndLevel("doubleYellow", 3, testclass);
+
+        testclass.up("[SC]");
+        Assertions.assertFalse(testclass.down(arg));
+        assertEqualColorAndLevel("[SC]", 3, testclass);
+
+        testclass.up("red");
+        Assertions.assertFalse(testclass.down(arg));
+        assertEqualColorAndLevel("red", 4, testclass);
+
+        testclass.up("end");
+        Assertions.assertFalse(testclass.down(arg));
+        assertEqualColorAndLevel("yellow", 5, testclass);
+    }
+
+    @Test
+    @DisplayName("Danger")
+    void downDanger() {
+        String arg = "danger";
+
+        Assertions.assertFalse(testclass.down(arg));
+        assertEqualColorAndLevel("", 0, testclass);
+
+        testclass.up("green");
+        Assertions.assertFalse(testclass.down(arg));
+        assertEqualColorAndLevel("green", 1, testclass);
+
+        testclass.up("blue");
+        Assertions.assertFalse(testclass.down(arg));
+        assertEqualColorAndLevel("blue", 1, testclass);
+
+        testclass.up("yellow");
+        Assertions.assertTrue(testclass.down(arg));
+        assertEqualColorAndLevel("green", 1, testclass);
+
+        testclass.up("doubleYellow");
+        Assertions.assertTrue(testclass.down(arg));
+        assertEqualColorAndLevel("green", 1, testclass);
+
+        testclass.up("[SC]");
+        Assertions.assertTrue(testclass.down(arg));
+        assertEqualColorAndLevel("green", 1, testclass);
+
+        testclass.up("red");
+        Assertions.assertTrue(testclass.down(arg));
+        assertEqualColorAndLevel("green", 1, testclass);
+
+        testclass.up("end");
+        Assertions.assertFalse(testclass.down(arg));
+        assertEqualColorAndLevel("yellow", 5, testclass);
+    }
+
+
+    @Test
+    void testToString() {
+    }
+
+    @Test
+    void getPostNumber() {
```

Comment:
Maybe rename this method as the name is kind of not saying what the method actual does.